lib/ethereumex/client/base_client.ex

```diff
@@ -165,9 +165,9 @@ defmodule Ethereumex.Client.BaseClient do
         "eth_call" |> request(params, opts)
       end
 
-      @spec eth_estimate_gas(map(), binary(), keyword()) :: {:ok, binary()} | error
-      def eth_estimate_gas(transaction, block \\ "latest", opts \\ []) do
-        params = [transaction, block]
+      @spec eth_estimate_gas(map(), keyword()) :: {:ok, binary()} | error
+      def eth_estimate_gas(transaction, opts \\ []) do
+        params = [transaction]
```

Comment:
I think it accepts two parameters

Reply:
While that's what I got after reading Ethereum JSON RPC wiki, I can't find any examples of people using it like that - every single example I found doesn't specify the block type. Also, after removing the block parameter, the related test passed. 

With current implementation, I keep getting `{:error, %{"code" => -32602, "message" => "too many arguments, want at most 1"}}`

If I misunderstood the spec, would really appreciate if you could comment on that.

Reply:
ok. I'll look into it on the weekend